src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java

```diff
@@ -78,6 +89,25 @@
      */
     public Headers() {map = new HashMap<>(32);}
 
+    /**
+     * Creates a mutable {@code Headers} from the given {@code headers} with
+     * the same header names and values.
+     *
+     * @param headers a map of header names and values
+     * @throws NullPointerException if {@code headers} or any of its names or
+     *                              values are null, or if any value contains
+     *                              null.
+     * @since 18
+     */
+    public Headers(Map<String,List<String>> headers) {
+        Objects.requireNonNull(headers);
+        var h = headers.entrySet().stream()
+                .collect(Collectors.toUnmodifiableMap(
+                        Entry::getKey, e -> new LinkedList<>(e.getValue())));
```

Comment:
I wonder, what the reason of  `LinkedList` usages here?

Reply:
Hi Andrey, IIRC from when I reviewed this code the current implementation of `Headers` already uses `LinkedList` in other places - so this preserves the concrete list implementation class that `Headers` uses (see `Headers::add`). Not that it matters much - but if we wanted to replace `LinkedList` with `ArrayList`  I believe we should do it consistently in this class - and probably outside of the realm of this JEP.

Reply:
That's right, it's due to the pre-existing implementation.